lib/Sema/CSGen.cpp

```diff
@@ -1152,7 +1152,11 @@ namespace {
       // `_ = nil`, let's diagnose it here because solver can't
       // attempt any types for it.
       auto *parentExpr = CS.getParentExpr(expr);
-      if (parentExpr && isa<ParenExpr>(parentExpr))
+      
+      // Loop until you have an expr without parentheses 
+      // so if you have (nil), ((nil)), (((nil))), etc...
+      // you get `nil`
```

Comment:
This comment is misleading, it doesn't actually look down, it looks up, so if there is `(nil)` `parentExpr` would actually be `nullptr`, in cases like `_ = (nil)` it would point to `AssignExpr` etc.

Reply:
So I should remove the (nil) from the comment or do you think the comment isn't really helpful at all and I should remove it enirely?

Reply:
I mean comment is incorrect it makes it look like the loop goes down in the AST while it actually goes up. So it either has to be adjusted to reflect reality (preferred) or be removed.